src/hooks/useAsset/useAsset.tsx

```diff
@@ -1,120 +1,60 @@
-import isEqual from 'lodash/isEqual'
-import { useCallback, useEffect, useMemo, useRef, useState } from 'react'
+import { isEqual } from 'lodash'
+import { useEffect, useState } from 'react'
 
-import { AllChannelFieldsFragment, AssetAvailability, VideoFieldsFragment } from '@/api/queries'
-import { AssetType } from '@/hooks'
 import { useStorageProviders } from '@/providers'
-import { createStorageNodeUrl } from '@/utils/asset'
 import { Logger } from '@/utils/logger'
 
-type UseAssetDataArgs =
-  | { entity?: VideoFieldsFragment | null; assetType: AssetType.THUMBNAIL | AssetType.MEDIA }
-  | {
-      entity?: Partial<AllChannelFieldsFragment> | null
-      assetType: AssetType.COVER | AssetType.AVATAR
-    }
-
-type UseAssetData = {
-  url?: string
-  error: ErrorEvent | null
-  isLoading?: boolean
-}
+import { getAssetUrl, readAssetData, testAssetDownload } from './helpers'
+import { AssetData, UseAsset } from './types'
 
-type UseAsset = ({ entity, assetType }: UseAssetDataArgs) => UseAssetData
-
-export const useAsset: UseAsset = ({ entity, assetType }) => {
+export const useAsset: UseAsset = ({ entity, assetType }, opts = {}) => {
   const { getStorageProvider } = useStorageProviders()
   const [error, setError] = useState<ErrorEvent | null>(null)
-  const [isLoading, setIsLoading] = useState<boolean | undefined>(undefined)
+  const [isLoading, setIsLoading] = useState<boolean>(true)
   const [url, setUrl] = useState<string | undefined>(undefined)
-  const cachedAssetData = useRef({})
+  const [cachedAssetData, setCachedAssetData] = useState<AssetData | null | undefined>(undefined)
+
+  const assetData = entity ? readAssetData(entity, assetType) : null
+  const assetDataNotChanged = entity ? isEqual(assetData, cachedAssetData) : false
 
   useEffect(() => {
-    if (error) {
-      Logger.error(`Failed to load ${assetType}`, error)
+    if (!entity || assetDataNotChanged || opts.skip) {
```

Comment:
I could be wrong, but do we actually need `opts.skip`? I removed that from the code and didn't notice any difference, everything should work just fine. Tested updating thumbnails and switching tabs, with no problems

Reply:
Yeah seems it wasn't doing us any good, removed